### PR TITLE
Auth url

### PR DESCRIPTION
**Pull request by mholecy**

### CODE REVIEW

File: sdk/src/main/java/com/sygic/travel/sdk/Sdk.kt [hunk 1]
```diff
@@ -34,4 +40,10 @@ class Sdk(clientId: String, xApiKey: String, context: Context) {
 		get() {
 			return kodein.instance()
 		}
+
+	fun reloadAuthorizationServerUrl(authServerUrl: String) {
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/Sdk.kt]: Probably correct solution will be to make it a constructor dependency. 
Also, for our usecase, we would create two Kodeins immidiately.

File: sdk/src/main/java/com/sygic/travel/sdk/auth/service/AuthServiceUrlHolder.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.sygic.travel.sdk.auth.service
+
+object AuthServiceUrlHolder {
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/auth/service/AuthServiceUrlHolder.kt]: This is definitely not a good solution. 
- Object imo this is wrong
- there should be interface
- with default static implementation

Also, not sure if the name of the class should be so specific.  Also, it is not about auth url.
I have told you it should configure both urls!.

File: sdk/src/main/java/com/sygic/travel/sdk/di/sygicAuthApiModule.kt [hunk 3]
```diff
@@ -24,7 +25,7 @@ internal val sygicAuthApiModule = Kodein.Module {
 	bind<Retrofit>("sygicAuthApiRetrofit") with singleton {
 		Retrofit.Builder()
 			.client(instance("sygicAuthHttpClient"))
-			.baseUrl("https://auth.sygic.com")
+			.baseUrl(AuthServiceUrlHolder.getAuthServerUrl())
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/di/sygicAuthApiModule.kt]: No nependency injection, really?

File: sdk/src/main/java/com/sygic/travel/sdk/Sdk.kt (hunk 1 above)
Comment (hrach) [reply sdk/src/main/java/com/sygic/travel/sdk/Sdk.kt]: Please, read the configuration urls from the object and then put them into kondein.

File: sdk/src/main/java/com/sygic/travel/sdk/Sdk.kt [hunk 4]
```diff
@@ -12,8 +12,17 @@ import com.sygic.travel.sdk.tours.facade.ToursFacade
 /**
  * Provides public methods for requesting API.
  */
-class Sdk(clientId: String, xApiKey: String, context: Context) {
-	private var kodein: Kodein = KodeinSetup.setupKodein(clientId, xApiKey, context)
+class Sdk(
+	clientId: String,
+	xApiKey: String,
+	context: Context,
+	urlConfig: UrlConfig
```
Comment (hrach) [inline sdk/src/main/java/com/sygic/travel/sdk/Sdk.kt]: isn't possible to do something like urlConfig = UrlConfig {}?

Comment (mholecy) [reply sdk/src/main/java/com/sygic/travel/sdk/Sdk.kt]: No.

Comment (mholecy) [reply sdk/src/main/java/com/sygic/travel/sdk/Sdk.kt]: Neither `urlConfig = object : UrlConfig {}`

Comment (hrach) [reply sdk/src/main/java/com/sygic/travel/sdk/Sdk.kt]: you should try harder... `urlConfig: UrlConfig = object : UrlConfig {}` seems working ;)

Comment (mholecy) [reply sdk/src/main/java/com/sygic/travel/sdk/Sdk.kt]: Some Kotlin magic :D

**Pull request merged**